va/caa_test.go

```diff
@@ -518,57 +521,107 @@ func TestCAALogging(t *testing.T) {
 	}
 }
 
+type caaCheckFuncRunner func(context.Context, *ValidationAuthorityImpl, *vapb.IsCAAValidRequest) (*vapb.IsCAAValidResponse, error)
```

Comment:
I really wish Go would let us declare a method type instead of just a func type, so you could say something like
```go
type caaCheckFunc (*ValidationAuthorityImpl) func(context.Context, *vapb.IsCAAValidRequest) (*vapb.IsCAAValidResponse, error)
```
and then not have to bother with the wrapper funcs below